extensions/oidc-db-token-state-manager/runtime/src/main/java/io/quarkus/oidc/db/token/state/manager/runtime/OidcDbTokenStateManagerInitializer.java

```diff
@@ -0,0 +1,145 @@
+package io.quarkus.oidc.db.token.state.manager.runtime;
+
+import static io.quarkus.oidc.db.token.state.manager.runtime.OidcDbTokenStateManager.now;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import jakarta.enterprise.event.Observes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.sqlclient.Pool;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+
+public class OidcDbTokenStateManagerInitializer {
+
+    private static final Logger LOG = Logger.getLogger(OidcDbTokenStateManagerInitializer.class);
+    private static final String FAILED_TO_CREATE_DB_TABLE = "unknown reason, please report the issue and create table manually";
+    /**
+     * Extra 30 seconds before we delete expired tokens.
+     */
+    private static final long EXPIRED_EXTRA_GRACE = 30;
+
+    void initialize(@Observes StartupEvent event, OidcDbTokenStateManagerRunTimeConfig config, Vertx vertx, Pool pool,
+            OidcDbTokenStateManagerInitializerProperties initializerProps) {
+        if (config.createDatabaseTableIfNotExists()) {
+            createDatabaseTable(pool, initializerProps.createTableDdl, initializerProps.supportsIfTableNotExists);
+        }
+        periodicallyDeleteExpiredTokens(vertx, pool, config.deleteExpiredDelay().toMillis());
+    }
+
+    private static void periodicallyDeleteExpiredTokens(Vertx vertx, Pool pool, long delayBetweenChecks) {
+        vertx
+                .setPeriodic(5000, delayBetweenChecks, new Handler<Long>() {
+
+                    private final AtomicBoolean deleteInProgress = new AtomicBoolean(false);
+
+                    @Override
+                    public void handle(Long aLong) {
+                        if (deleteInProgress.compareAndSet(false, true)) {
+
+                            final long deleteExpiresIn = now() - EXPIRED_EXTRA_GRACE;
```

Comment:
I'm not sure it works exactly as expected. The session_max_age which was saved in the record is a relative value not related to the epoch (ex, base max age there is a difference between ID token `issued at` and `expires in` claim values). So this DELETE check will be true in most cases. 
I believe we should save `now() + session_max_age`  for this DELETE check to work ?

Reply:
Session max age is `Indicates the number of seconds until the cookie expires.`, in the database it is stored as `expires_in=maxAgeInSeconds + epochInSeconds` (meta code). Therefore delete check is correct in my eyes, for this way, I can run DELETE statement without handling differences of time data types between databases. 

And now, I added `EXPIRED_EXTRA_GRACE` constant based on this comment on yours _The expiry should probably be calculated the same way the q_session coookie lifespan is calculated plus add a few mins to it just in case._ https://github.com/quarkusio/quarkus/issues/13239#issuecomment-932938148

Reply:
@michalvavrik Oh I missed you were saving it as ` now() + maxAge` sure, make sense